daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -245,6 +246,8 @@ object TransactionCoder {
       enclosingVersion: TransactionVersion,
       nodeId: Nid,
       node: GenNode[Nid, Cid],
+      disableVersionCheck: Boolean =
+        false, //true allows encoding of bad protos (for testing of decode checks)
```

Comment:
I am a bit uncomfortable with adding this kind of logic only for testing. 
This can be void by tweaking the proto directly as suggested below.